build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/leafTask.ts

```diff
@@ -565,4 +572,31 @@ export abstract class LeafWithFileStatDoneFileTask extends LeafWithDoneFileTask
 			return undefined;
 		}
 	}
+
+	private async getHashDoneFileContent(): Promise<string | undefined> {
```

Comment:
I'm partial to not adding this now and keep it in a branch, but no strong feelings about it.

Reply:
It definitely needs more comments.
I have questions about the whole system. Hopefully I can find enough context in some other files (that aren't changed here).